boards/common/nucleo/Makefile.include

```diff
@@ -1,5 +1,38 @@
-# include nucleo common serial configuration
-include $(RIOTBOARD)/common/nucleo/Makefile.include.serial
+# export this module and its includes
+USEMODULE += boards_common_nucleo
+INCLUDES  += -I$(RIOTBOARD)/common/nucleo/include
 
-# add the common header files to the include path
-INCLUDES += -I$(RIOTBOARD)/common/nucleo/include
+# configure the serial terminal
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+# all Nucleo boards have an on-board ST-link v2-1 adapter
+export DEBUG_ADAPTER ?= stlink
+export STLINK_VERSION ?= 2-1
+
+# select OpenOCD configuration depending on CPU type
+ifeq (,$(OPENOCD_CONFIG))
+  ifeq ($(CPU),stm32f0)
```

Comment:
how about
```export OPENOCD_CONFIG := $(RIOTBOARD)/common/nucleo/dist/$(patsubst stm32%,openocd-%.cfg,$(CPU))```.

And please add an error case, e.g., err out on ```if $(,$(filter stm32%,$(CPU))```

Reply:
sounds good. Ok if I do it in a separate PR?

Reply:
done #8391: https://github.com/RIOT-OS/RIOT/pull/8391/commits/9bbd78d0d8b945b517f459680a5dcd52bb78f546

Reply:
ok